perceval/components/generic_interferometer.py

```diff
@@ -0,0 +1,193 @@
+        if self._has_input_phase_layer:
+            p_idx += self.m
+        return p_idx
+
+    @staticmethod
+    def _compute_insertion_depth(start_col: int, m: int, param_count: int) -> int:
+        depth = 0
+        cc = 0
+        k = 0
+        while k < param_count:
+            depth += 1
+            k += 2*(m//2) if (start_col+cc)%2 == 0 else 2*((m-1)//2)
+            cc += 1
+        return depth
+
+    def set_param_list(self, param_list: List[float], top_left_pos: Tuple[int, int], m: int):
+        """Insert parameters value starting from a given position in the interferometer.
+
+        This method is designed to work on rectangular interferometers
+
+        :param param_list: List of numerical values for the parameters
+        :param top_left_pos: Starting position of the insertion (column#, row#). Position is handled MZI-wise (i.e.
+        (0,0) starts inserting values on the top-left-most MZI of the interferometer whereas (1,0) starts on top of the
+        2nd MZI column). The optional phase layer is ignored in the position handling.
+        :param m: Mode count on where to insert the parameter values
+        """
+        col, lin = top_left_pos
+        depth = self._compute_insertion_depth(col, m, len(param_list))
+        if col < 0 or col+depth-1 >= self.m:
+            raise ValueError(f"Invalid param list width, expected interval in [0,{self.m}], got [{col},{col+depth-1}]")
+        if lin < 0 or lin+m >= self.m:
+            raise ValueError(f"Invalid param list height, expected interval in [0,{self.m}], got [{lin},{lin+m}]")
+        if self._shape != InterferometerShape.RECTANGLE:
+            warn(f"set_param_list was designed for rectangular interferometer")
+
+        even_mode_count = self.m % 2 == 0
+        even_col_size = self.m - (self.m % 2)
+        odd_col_size = even_col_size - 2 if even_mode_count else even_col_size
+        self_params = self.get_parameters()
+        cc = 0  # current col
+        k = 0
+        while k < len(param_list):
+            start = self._find_param_index(col+cc, lin, even_col_size, odd_col_size)
+            param_count = 2*(m//2) if (col+cc) % 2 == 0 else 2*((m-1)//2)
+            for j in range(param_count):
+                self_params[start+j].set_value(param_list[k])
+                k += 1
+                if k == len(param_list):
+                    break
+            cc += 1
+
+    def set_params_from_other(self, other: Circuit, top_left_pos: Tuple[int, int]):
+        """Retrieve parameter value from another interferometer
+
+        :param other: Another interferometer
+        :param top_left_pos: Starting position of the insertion. See full description in `set_param_list`
+        """
+        if not other.defined:
+            raise ValueError("Cannot copy parameters from a circuit which isn't fully defined")
+        param_list = [float(p) for p in other.get_parameters()]
+        self.set_param_list(param_list, top_left_pos, other.m)
```

Comment:
Consider adding a __repr__ method to display some infos in debug mode (since the interferometer cannot be displayed)

    def __repr__(self):
        return f"shape:{self._shape}, depth:{self._depth}, depth_per_mode:{self._depth_per_mode}, has_input_phase_layer:{self._has_input_phase_layer}"

Reply:
Done, with minimal info